Blue Ocean Support

grfullerton opened the pull request:
So I've been making some progress in getting the Pull Request tab of Blue Ocean to work properly for this plugin (see #29).   I've been able to get it mostly working, however **I did have to make some changes that affect the classic UI as well**.

#29 seems to be caused due to spaces in title field.  So, I adopted the pattern used by Github: naming MRs as "MR-#". So the classic UI does lose the more descriptive name. The other changes shouldn't impact existing behavior to my knowledge.

grfullerton commented:
@Argelbargel I'd like your input on the change regarding the name of MRs in the classic Jenkins UI.  Although its in line with how github-branch-source behaves, it is change that could be seen as a regression to current users of this plugin that aren't using blue ocean.

witokondoria commented:
naming MRs with just the number, whilst keeping the title as description will presumably solve MRs renames kicking unnecesary builds, isn't it?

acarmel commented:
Any updates?

StefanKoenen commented:
please?

guillaumelecerf commented:
I tested this patch and it's working for unmerged branches, but the URL is still bad for merged branches, i.e. it contains `(merged)` which still breaks Blue Ocean.

StefanKoenen commented:
@guillaumelecerf Please try my fork, I'm using it for 2 months now. https://github.com/StefanKoenen/gitlab-branch-source-plugin

guillaumelecerf commented:
@StefanKoenen : what plugin settings you are using for your jenkins projects?

StefanKoenen commented:
@guillaumelecerf 
![image](https://user-images.githubusercontent.com/1442819/55345077-f39d7e00-54a6-11e9-8f20-eeb72a7fea9b.png)
![image](https://user-images.githubusercontent.com/1442819/55345145-1fb8ff00-54a7-11e9-9bfd-17c653c531a9.png)

And in Gitlab add a webrequest hook to: "my-jenkins-instance.com/job/{jenkins-project-name}/job/{gitlab-namespace/project}/build"

guillaumelecerf commented:
And what abour "gitlab connection" -> "advanced" ?
Because I can see that you're using the "classical" build strategies, but what about the gitlab specific build strategies?

StefanKoenen commented:
@guillaumelecerf 
This is in the advanced section: 
![image](https://user-images.githubusercontent.com/1442819/55408225-d973a680-555f-11e9-906e-82ce0fed66a4.png)

I Don't understand what you mean with "classical" vs "gitlab specific" build strategies?

guillaumelecerf commented:
Plz loog at => https://github.com/Argelbargel/gitlab-branch-source-plugin/pull/104

guillaumelecerf commented:
@StefanKoenen : there are currently 2 build strategies settings: the one coming with the gitlab plugin, in the advanced section, and the one a bit lower in the config page (aka "classical" in my mind).
Hope it's clearer

grfullerton commented:
@guillaumelecerf I just pushed a commit that _should_ fix the merged MRs in blueocean, but I haven't tested it.  

Does #104 fix the issue where "Build merged with base branch" always is checked, even after unchecking and saving? Because that's literally the only bug I've encountered that I haven't been able to fix for my usage of this plugin.

I probably need to remove 2974c7a from this PR since it doesn't actually fix that problem.  I think 0e92548 probably is in the same boat, I'm running a different fork on my own jenkins without that change and haven't had any issues with duplicate branches

grfullerton commented:
I removed the commits that were unrelated/unnecessary for getting blueocean working. @guillaumelecerf let me know if my fix for merged branches works.  If it does, I'll remove the [WIP] unless there are other remaining issues specific to blueocean.

guillaumelecerf commented:
@grfullerton : 
> Does #104 fix the issue where "Build merged with base branch" always is checked, even after unchecking and saving? Because that's literally the only bug I've encountered that I haven't been able to fix for my usage of this plugin.

Yes, that's exactly that :)

guillaumelecerf commented:
@grfullerton :
> I removed the commits that were unrelated/unnecessary for getting blueocean working. @guillaumelecerf let me know if my fix for merged branches works. If it does, I'll remove the [WIP] unless there are other remaining issues specific to blueocean.

I did the same fix in my personal fork, and it's working fine, so go for removing the WIP status.

guillaumelecerf commented on src/main/java/argelbargel/jenkins/plugins/gitlab_branch_source/SourceActions.java:
```diff
@@ -70,10 +71,11 @@
         actions.add(new GitLabSCMPublishAction(head, source.getSourceSettings()));
 
         Action linkAction;
-
```
maybe should stay

grfullerton replied:
fixed

guillaumelecerf commented on src/main/java/argelbargel/jenkins/plugins/gitlab_branch_source/heads/GitLabSCMMergeRequestHead.java:
```diff
@@ -32,26 +32,32 @@
         this.merge = merge;
     }
 
+    public String getDescription() {
```
can not figure out where this function is used

Argelbargel replied:
Did not look in the code yet, but i think it is (was?) used as the job's description in Jenkins below the job name

grfullerton replied:
Its used in `SourceActions.java` on line 78: 
```
actions.add(createHeadMetadataAction(((GitLabSCMMergeRequestHead) head).getDescription(), ((GitLabSCMMergeRequestHead) head).getSource(), null, linkAction.getUrlName()));
```

grfullerton commented:
I addressed all the changes/discussions @guillaumelecerf had.  I've also tested and verified my fix for merged branches in blue ocean.  I removed the WIP tag, as I think this is now good to go.

guillaumelecerf commented:
LGTM

guillaumelecerf commented:
Please @Argelbargel review

dastanko commented:
Any plans to release this fix?

Argelbargel commented:
@dastanko: i've built 0.7.8-dev yesterday containing  @grfullerton's fixes
